deepchem/models/torch_models/layers.py

```diff
@@ -6164,3 +6164,182 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         output = H_out * T_out + x * (1 - T_out)
 
         return output
+
+
+def cosine_dist(x, y, eps=1e-8):
+    """Computes the inner product (cosine similarity) between two tensors.
+
+    This assumes that the two input tensors contain rows of vectors where
+    each column represents a different feature. The output tensor will have
+    elements that represent the inner product between pairs of normalized vectors
+    in the rows of `x` and `y`. The two tensors need to have the same number of
+    columns, because one cannot take the dot product between vectors of different
+    lengths. For example, in sentence similarity and sentence classification tasks,
+    the number of columns is the embedding size. In these tasks, the rows of the
+    input tensors would be different test vectors or sentences. The input tensors
+    themselves could be different batches. Using vectors or tensors of all 0s
+    should be avoided.
+
+    Methods
```

Comment:
Let's remove this 'Methods' section and make it a second paragraph to the description